### PR TITLE
refactor dubbo-samples-version

### PR DESCRIPTION
**Pull request by FinalT**

### CODE REVIEW

**Review by chickenlj — COMMENTED** [review]
AddressListener and Transport tests are missed in the new version implementation.

**Comment by chickenlj** [comment]
Newly added files are missing ASF headers

**Comment by FinalT** [comment]
Why annotations don't identify spi

**Review by AlbumenJ — APPROVED** [review]

File: 2-advanced/dubbo-samples-version/case-versions.conf [hunk 1]
```diff
@@ -20,6 +20,6 @@
 # Supported component versions of the test case
 
 # Spring app
-dubbo.version=2.7*, 3.*
+dubbo.version=2.7*, 3.*, !3.1.8*, !3.2.0-beta.6*
```
Comment (AlbumenJ) [inline 2-advanced/dubbo-samples-version/case-versions.conf:23]: Why ignore this version

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: 2-advanced/dubbo-samples-version/case-versions.conf (hunk 1 above)
Comment (FinalT) [reply 2-advanced/dubbo-samples-version/case-versions.conf:23]: https://github.com/apache/dubbo/issues/11766

File: 2-advanced/dubbo-samples-version/case-versions.conf [hunk 2]
```diff
@@ -20,6 +20,6 @@
 # Supported component versions of the test case
 
 # Spring app
-dubbo.version=2.7*, 3.*
+dubbo.version=2.7*, 3.*, !3.1.8*, !3.2.0-beta.6* ,!3.1.9* ,!3.2.0-beta.7*
```
Comment (AlbumenJ) [inline 2-advanced/dubbo-samples-version/case-versions.conf:23]: What is the purpose of add these versions

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: 2-advanced/dubbo-samples-version/case-versions.conf (hunk 2 above)
Comment (FinalT) [reply 2-advanced/dubbo-samples-version/case-versions.conf:23]: because of https://github.com/apache/dubbo/issues/11766

**Pull request closed without merging**